scRNA-seq-advanced/setup/intro-setup.Rmd

```diff
@@ -0,0 +1,340 @@
+dim(filtered_sce)
+```
+
+
+
+### Normalize RNA and ADT:
+
+
+```{r}
+# Normalize RNA
+qclust <- scran::quickCluster(filtered_sce)
+filtered_sce <- scran::computeSumFactors(filtered_sce, clusters = qclust)
+norm_sce <- scater::logNormCounts(filtered_sce)
+
+# Normalize ADT
+
+# Calculate baseline:
+# Try out control-based normalization here 
+#  http://bioconductor.org/books/3.16/OSCA.advanced/integrating-with-protein-abundance.html#control-based-normalization
+control_sf <- scater::librarySizeFactors(altExp(norm_sce), subset_row=controls)  # `controls` was previously defined during filtering
+sum(control_sf == 0)/length(control_sf) # _HOLY WOW_ - virtually everything is 0.......
+
+
+# Go for median instead, BUT TODO: would we want to exclude control here during baseline calculation? 
+baseline <- DropletUtils::ambientProfileBimodal(altExp(norm_sce)) # !! currently this does not exclude controls
+median_sf <- scater::medianSizeFactors(altExp(norm_sce), reference = baseline)
+sum(median_sf == 0) # none are 0 here, soooo let's do this?
+
+# Normalize with the median size factors
+altExp(norm_sce) <- scater::logNormCounts(altExp(norm_sce), 
+                                          size.factors = median_sf)
+
+```
+
+### Behold, the filtered and normalized SCE:
+
+```{r}
+norm_sce
+
+altExp(norm_sce)
+```
+
+## Dimension reduction
+
+HVGs:
+```{r}
+num_genes <- 5000 # 2000? 
+
+gene_variance <- scran::modelGeneVar(norm_sce)
+hvg_list <- scran::getTopHVGs(gene_variance,
+                               n = num_genes)
+```
+
+PCA:
+```{r}
+# Do we want to set the number of PCs?
+norm_sce <- scater::runPCA(norm_sce,
+                           subset_row = hvg_list)
+```
+
+UMAP:
+```{r I map you map we all map for MapMap}
+# Perform graph clustering first
+nn_k <- 15
+nn_clusters <- clusterRows(
+  reducedDim(norm_sce, "PCA"), 
+  NNGraphParam(k = nn_k, 
+               cluster.fun = "walktrap")
+)
+
+# pop back into sce
+norm_sce$nn_clusters <- factor(nn_clusters)
+
+# Add in UMAPs and plot with colors
+norm_sce <- scater::runUMAP(norm_sce,
+                            dimred = "PCA")
+scater::plotReducedDim(norm_sce, "UMAP", colour_by = "nn_clusters")
+```
+
+## Cell type annotation
+
+We can introduce annotation with `SingleR`, and it might be a good idea to show how the results are highly influenced by the reference. 
+For now, I'm just using the `HumanPrimaryCellAtlasData` reference as a starting place.
+
+
+```{r}
+ref_data <- celldex::HumanPrimaryCellAtlasData(ensembl = TRUE) # pick one for now to start
+preds <- SingleR::SingleR(
+  test = norm_sce, 
+  ref = ref_data,
+  labels = ref_data$label.main # "coarse-grained" labels
+)
+preds
+
+# We can place these into the sce
+norm_sce$celltype_predictions <- preds$pruned.labels
+
+# **heavily** biased towards monocytes and T-cells
+celltype_counts <- table(preds$pruned.labels) %>%
+  as.data.frame() %>%
+  dplyr::rename(
+    celltype = Var1, 
+    n = Freq
+  ) %>%
+  dplyr::arrange(-n)
```

Comment:
Doing this a bit more tidily?
```suggestion
celltype_counts <- preds %>%
  as.data.frame() %>%
  dplyr::count(pruned.labels) %>%
  dplyr::arrange(desc(n))
```

I also wonder if we might want to start with `colData(norm_sce)` for consistency... since we put the results in there already.